File: Integration Examples/eAPI v1.6/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/PayReq.cs
using System.Text;
using Newtonsoft.Json;

namespace CsobGatewayClientExample.Communication.DataObjects
{
    public class PayReq : SignBase, IBaseRequest
    {
        [JsonProperty("merchantId")]
        public string MerchantId { get; set; }

        [JsonProperty("payId")]
        public string PayId { get; set; }

        public PayReq()
        {

        }

        public PayReq(string merchantId, string payId)
        {
            this.MerchantId = merchantId;
            this.PayId = payId;
        }

        public virtual string ToSign()
        {
            StringBuilder sb = new StringBuilder();
            Add(sb, MerchantId);
            Add(sb, PayId);
            Add(sb, DateTime);
            DeleteLast(sb);
            return sb.ToString();
        }
    }
}

File: Integration Examples/eAPI v1.8/java/eAPI-v18-connector/src/main/java/cz/monetplus/mips/eapi/v18/connector/entity/PayReq.java
package cz.monetplus.mips.eapi.v18.connector.entity;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement
@XmlAccessorType(XmlAccessType.FIELD)
public class PayReq extends SignBase {

	private static final long serialVersionUID = -3825192932302805075L;
	
	@XmlElement(nillable=false)
	public String merchantId;

	@XmlElement(nillable=false)
	public String payId;

	public PayReq() {
		
	}

	public PayReq(String merchantId, String payId) {
		this.merchantId = merchantId;
		this.payId = payId;
	}

	@Override
	public String toSign() {
		StringBuilder sb = new StringBuilder(); 
		add(sb, merchantId);
		add(sb, payId);
		add(sb, dttm);
		deleteLast(sb);
		return sb.toString();
	}

}
